### DISCUSSION TITLE
Server Error 500 -- Firefly Import again

### DISCUSSION THREAD

**Discussion by JuIez** [firefly-iii/firefly-iii#7859] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [x] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I have been attempting to set up the Firefly Importer, but I keep encountering a persistent 500 Server Error. Despite reviewing both the open and closed threads here, I haven't been able to find a solution to my problem.

Interestingly, I don't encounter any issues when using the Public Importer (https://data-importer.firefly-iii.org/). Unfortunately, after several hours of experimentation, I have been unable to resolve this issue.

### Debug information

Debug information generated at 2023-08-16 18:20:10 Europe/Berlin for Firefly III Data Importer version **1.3.3**.

| Scope | Version |
| --- | --- |
| FIDI | 1.3.3 |
| Build | 234, 16-07-2023 04:28:08 UTC |
| PHP | 8.2.8 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | true |
| System TZ | Europe/Berlin |
| Browser TZ | Europe/Berlin |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| BCscale | 12 |
| Trusted proxies (.env) | ** |

| User info | Value |
| --- | --- |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0 |

### Expected behaviour

I expected to be able to successfully set up the Firefly Importer without encountering a 500 Server Error.

### Steps to reproduce

1. Set up Firefly Importer following the provided instructions.
2. Encounter persistent 500 Server Error during the setup process.
3. Tried various troubleshooting steps, but none have resolved the issue.

### Additional info

Docker Logs Firefly-app:
[Wed Aug 16 18:13:40.771080 2023] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.56 (Debian) configured -- resuming normal operations
[Wed Aug 16 18:13:40.771324 2023] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'
127.0.0.1 - - [16/Aug/2023:18:14:21 +0200] "GET /health HTTP/1.1" 200 1776 "-" "Firefly III Health Checker/2.1"
92.118.39.81 - - [16/Aug/2023:18:14:47 +0200] "CONNECT myipb1a.mrrage.xyz:80 HTTP/1.1" 404 3678 "-" "-"
127.0.0.1 - - [16/Aug/2023:18:15:21 +0200] "GET /health HTTP/1.1" 200 1776 "-" "Firefly III Health Checker/2.1"
127.0.0.1 - - [16/Aug/2023:18:16:21 +0200] "GET /health HTTP/1.1" 200 1776 "-" "Firefly III Health Checker/2.1"
10.20.0.201 - - [16/Aug/2023:18:16:28 +0200] "GET /oauth/authorize?client_id=55&redirect_uri=https%3A%2F%2Fmoney-import.******.net%2Fcallback&response_type=code&scope=&state=bPvI3b2gKVqx2T6RZvI5mZ9D6Lq6PUoytXa8u4aD&code_challenge=FeE-qREy_AVX55fvPajZ0MS_0PYW5nw3MD4Bp2EFl2U&code_challenge_method=S256 HTTP/1.1" 200 3607 "https://money-import.******.net/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0"
10.20.0.201 - - [16/Aug/2023:18:16:29 +0200] "POST /oauth/authorize HTTP/1.1" 302 2660 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0"
10.20.0.201 - - [16/Aug/2023:18:16:36 +0200] "GET /profile HTTP/1.1" 200 9636 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0"
10.20.0.201 - - [16/Aug/2023:18:16:37 +0200] "GET /v1/jscript/variables?ext=.js&v=6.0.20 HTTP/1.1" 200 3691 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0"
10.20.0.201 - - [16/Aug/2023:18:16:37 +0200] "GET /oauth/personal-access-tokens HTTP/1.1" 200 3028 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0"
10.20.0.201 - - [16/Aug/2023:18:16:37 +0200] "GET /oauth/clients HTTP/1.1" 200 2751 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0"
10.20.0.201 - - [16/Aug/2023:18:16:37 +0200] "GET /oauth/scopes HTTP/1.1" 200 2456 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0"
10.20.0.201 - - [16/Aug/2023:18:16:37 +0200] "GET /oauth/tokens HTTP/1.1" 200 2456 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0"
193.35.18.33 - - [16/Aug/2023:18:17:14 +0200] "CONNECT google.com:443 HTTP/1.1" 404 3674 "-" "Go-http-client/1.1"
127.0.0.1 - - [16/Aug/2023:18:17:22 +0200] "GET /health HTTP/1.1" 200 1776 "-" "Firefly III Health Checker/2.1"

Docker-Logs Firefly-Importer:
#31 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#32 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#33 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#35 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#36 /var/www/html/vendor/laravel/framework/src/Illuminate/Http/Middleware/HandleCors.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#37 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Http\\Middleware\\HandleCors->handle()
#38 /var/www/html/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#39 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Http\\Middleware\\TrustProxies->handle()
#40 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#41 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(175): Illuminate\\Pipeline\\Pipeline->then()
#42 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(144): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#43 /var/www/html/public/index.php(76): Illuminate\\Foundation\\Http\\Kernel->handle()
#44 {main}
"} 
10.20.0.201 - - [16/Aug/2023:18:20:08 +0200] "GET /callback?code=def50200d1a0aa8b7b01d2d86e96ba043420831a31ee5f341a02a74832eb585c2650871f9cd5dbfea6b893a169a2b438458200ac8d2cce12b0eb23fcf13070348528362f8ece941ce820e0de181ee1d77856828a753c5761947a1c08d57af1db3980ce3f2f65c875238d62b5f269c30db38d9b0be8ef2a871f02548bfcff428e6ff91c67f2ad895cdabefdfe3cd2b892656d9980e60a59be97a0b2c3f241195acacee898424c6fb76827d3ece4f81d2fe5e94e1419d65511fcb7a6d0ef376571e72915ee201105f6b3b5447304986ed27d98c8599e54923282f2c39a1c71bb329f2c2af66b3fe9375e27dfa4a44677451c9063dd0ff234c214b55f1b5ae6d9da58ea9b87fcb9757dce282537d9a96941423d121674e3dc9564dde208705ab7eb5f137122d3286a95b044276ddd69355d6967c813811c591884c0e22fcdcc5520ce91f1d0e5f6c843d2e8cb4ab83837905a9f123378ce07197c90499be4d6b1319cba33511b6fa15d20f653e872ec0c4ff07829a19a847ce2dda9f93c3f31ccab4935db38567a46c4a64d80ec606ed1f65757b04186b7baa590af&state=bPvI3b2gKVqx2T6RZvI5mZ9D6Lq6PUoytXa8u4aD HTTP/1.1" 500 3551 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0"
[2023-08-16 18:20:10] local.EMERGENCY: I am a EMERGENCY message.  
[2023-08-16 18:20:10] local.ALERT: I am a ALERT message.  
[2023-08-16 18:20:10] local.CRITICAL: I am a CRITICAL message.  
[2023-08-16 18:20:10] local.ERROR: I am a ERROR message.  
[2023-08-16 18:20:10] local.WARNING: I am a WARNING message.  
[2023-08-16 18:20:10] local.NOTICE: I am a NOTICE message.  
[2023-08-16 18:20:10] local.INFO: I am a INFO message.  
[2023-08-16 18:20:10] local.DEBUG: I am a DEBUG message.  
10.20.0.201 - - [16/Aug/2023:18:20:10 +0200] "GET /debug HTTP/1.1" 200 3495 "https://money-import.******.net/callback?code=def50200d1a0aa8b7b01d2d86e96ba043420831a31ee5f341a02a74832eb585c2650871f9cd5dbfea6b893a169a2b438458200ac8d2cce12b0eb23fcf13070348528362f8ece941ce820e0de181ee1d77856828a753c5761947a1c08d57af1db3980ce3f2f65c875238d62b5f269c30db38d9b0be8ef2a871f02548bfcff428e6ff91c67f2ad895cdabefdfe3cd2b892656d9980e60a59be97a0b2c3f241195acacee898424c6fb76827d3ece4f81d2fe5e94e1419d65511fcb7a6d0ef376571e72915ee201105f6b3b5447304986ed27d98c8599e54923282f2c39a1c71bb329f2c2af66b3fe9375e27dfa4a44677451c9063dd0ff234c214b55f1b5ae6d9da58ea9b87fcb9757dce282537d9a96941423d121674e3dc9564dde208705ab7eb5f137122d3286a95b044276ddd69355d6967c813811c591884c0e22fcdcc5520ce91f1d0e5f6c843d2e8cb4ab83837905a9f123378ce07197c90499be4d6b1319cba33511b6fa15d20f653e872ec0c4ff07829a19a847ce2dda9f93c3f31ccab4935db38567a46c4a64d80ec606ed1f65757b04186b7baa590af&state=bPvI3b2gKVqx2T6RZvI5mZ9D6Lq6PUoytXa8u4aD" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 OPR/100.0.0.0"
127.0.0.1 - - [16/Aug/2023:18:20:24 +0200] "GET /health HTTP/1.1" 200 1105 "-" "Firefly III Health Checker/2.0"

**Comment by JC5** [comment]
Sorry I don't see an error in the logs. Could you share that please? The stack trace is only partially present.

**Comment by JuIez** [comment]
Did u mean from the FIDI Debug or the whole log from the containers?

```
Truncated from this point <----|/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(175): Illuminate\\Pipeline\\Pipeline->then()
#57 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(144): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#58 /var/www/html/public/index.php(76): Illuminate\\Foundation\\Http\\Kernel->handle()
#59 {main}
"} 
[2023-08-16 18:20:08] local.DEBUG: Now at App\Http\Controllers\TokenController::callback  
[2023-08-16 18:20:08] local.ERROR: State according to session: ""  
[2023-08-16 18:20:08] local.ERROR: State returned in request : "bPvI3b2gKVqx2T6RZvI5mZ9D6Lq6PUoytXa8u4aD"  
[2023-08-16 18:20:08] local.ERROR: The "state" returned from your server doesn't match the state that was sent. {"exception":"[object] (App\\Exceptions\\ImporterErrorException(code: 0): The \"state\" returned from your server doesn't match the state that was sent. at /var/www/html/app/Http/Controllers/TokenController.php:74)
[stacktrace]
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): App\\Http\\Controllers\\TokenController->callback()
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(43): Illuminate\\Routing\\Controller->callAction()
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(259): Illuminate\\Routing\\ControllerDispatcher->dispatch()
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\\Routing\\Route->runController()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(799): Illuminate\\Routing\\Route->run()
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
#10 /var/www/html/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
#14 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Session\\Middleware\\StartSession->handle()
#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#19 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#20 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(798): Illuminate\\Pipeline\\Pipeline->then()
#21 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(777): Illuminate\\Routing\\Router->runRouteWithinStack()
#22 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(741): Illuminate\\Routing\\Router->runRoute()
#23 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(730): Illuminate\\Routing\\Router->dispatchToRoute()
#24 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(200): Illuminate\\Routing\\Router->dispatch()
#25 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#26 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#27 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#28 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
#29 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#30 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#31 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#32 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#33 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#35 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#36 /var/www/html/vendor/laravel/framework/src/Illuminate/Http/Middleware/HandleCors.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#37 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Http\\Middleware\\HandleCors->handle()
#38 /var/www/html/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#39 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\\Http\\Middleware\\TrustProxies->handle()
#40 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#41 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(175): Illuminate\\Pipeline\\Pipeline->then()
#42 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(144): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#43 /var/www/html/public/index.php(76): Illuminate\\Foundation\\Http\\Kernel->handle()
#44 {main}
"} 
[2023-08-16 18:20:10] local.EMERGENCY: I am a EMERGENCY message.  
[2023-08-16 18:20:10] local.ALERT: I am a ALERT message.  
[2023-08-16 18:20:10] local.CRITICAL: I am a CRITICAL message.  
[2023-08-16 18:20:10] local.ERROR: I am a ERROR message.  
[2023-08-16 18:20:10] local.WARNING: I am a WARNING message.  
[2023-08-16 18:20:10] local.NOTICE: I am a NOTICE message.  
[2023-08-16 18:20:10] local.INFO: I am a INFO message.  
[2023-08-16 18:20:10] local.DEBUG: I am a DEBUG message.  

```

**Comment by JC5** [comment]
```
[2023-08-16 18:20:08] local.ERROR: State according to session: ""  
[2023-08-16 18:20:08] local.ERROR: State returned in request : "bPvI3b2gKVqx2T6RZvI5mZ9D6Lq6PUoytXa8u4aD"  
[2023-08-16 18:20:08] local.ERROR: The "state" returned from your server doesn't match the state that was sent. {"exception":"[object] (App\\Exceptions\\ImporterErrorException(code: 0): The \"state\" returned from your server doesn't match the state that was sent. at /var/www/html/app/Http/Controllers/TokenController.php:74)
```

This is the problem, but it's not something I can fix. It's usually caused by missing or blocked cookies.

**Reply by JuIez** [reply]
Thank you for the insights. It appears I'll need to re-examine my NGINX Proxy Manager setup, although I haven't identified a solution yet. Any further recommendations are appreciated.

**Reply by apathub** [reply]
`[2023-08-19 00:39:05] production.ERROR: State according to session: ""  

[2023-08-19 00:39:05] production.ERROR: State returned in request : "uUXHlTgmLJ3POmUracIPY3VhJ87werYkqnmStMx0" 
 
[2023-08-19 00:39:05] production.ERROR: The "state" returned from your server doesn't match the state that was sent. {"exception":"[object] (App\\Exceptions\\ImporterErrorException(code: 0): The \"state\" returned from your server doesn't match the state that was sent. at /var/www/html/app/Http/Controllers/TokenController.php:74)`

I have exactly the same error and I have no clue how to tackle this

**Comment by JC5** [comment]
What is the address of the data importer and of Firefly III? Are you accessing it directly on the IP address (with port)? Some browsers have trouble setting cookies for IP addresses, and then Firefly III isn't able to save the state (which leads to this error).

**Reply by apathub** [reply]
So it was a networking issue. The main firefly installation was in a container connected to the host network. The firefly-importer was connected in a bridged docker network. I was accessing both using reversed proxy. Once I move both in the same subnet, and used the vanity url environment variable for my URL it worked like a charm!!

Thanks for the hint! I am going to try the importer out!

**Comment by repcsi** [comment]
Hi James,
I have the same problem. I'm just trying to test the app in docker and see how it works. app is running on localhost:80 and importer runs on localhost:81 Is there a way to add the authorization code somehow manually with the existing containers like you do it with the linux onedrive client for example?

**Reply by JC5** [reply]
https://docs.firefly-iii.org/data-importer/installation/configuration/#authentication

**Reply by technohiker** [reply]
I'm running into the same issue, but that link leads to a 404 page.
